python/comps.py

```diff
@@ -91,11 +97,30 @@ def get_comps(
 
     # Place observations in bins. Do this in a numba-accelerated function so
     # that we can make use of fast loops
-    observation_df["price_bin"] = _bin_by_price(
-        observation_df[["id", "predicted_value"]].values,
-        price_bin_indices.values
+    observation_matrix = observation_df[["id", "predicted_value"]].values
+    taichi_obs_ndarray = ti.ndarray(dtype=int, shape=observation_matrix.shape)
+    taichi_obs_ndarray.from_numpy(observation_matrix)
```

Comment:
Taichi cannot automatically convert numpy arrays to the data structures that it works with (called [fields](https://docs.taichi-lang.org/docs/field)) so we have to do that manually prior to passing them into the [kernel](https://docs.taichi-lang.org/docs/kernel_function).